### ISSUE TITLE
[Feature Request]: Logic for numeric range manual input 

### ISSUE THREAD

**Issue by lcd2yyz** [insightsengineering/teal.slice#231]
### Feature description

In addition to...
> - Can't set either lower or upper bound to 'e'. (allowed by browser, NA in Shiny server)
> - Can't specify lower bound > upper bound
from https://github.com/insightsengineering/teal.slice/pull/226

Some additional logic for the manual numeric range are outlined as follows
- Entered range must be a valid numerical value (but allow scientific notation like 2E6 or 2e-2)
- Entered range can be smaller/larger compared to the actual range of the filtered data or the slider (also mentioned here https://github.com/insightsengineering/teal.slice/issues/230)
- Allow user to specify one-side bound, eg. if left blank, consider it to be `-Inf` or `Inf`, effectively max of the filtered data.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

**Comment by asbates** [comment]
FYI if one of the inputs is empty, `NA` is returned to Shiny for that input. For example 'blank' to 3.4 will return `c(NA, 3.4)`. Because of this, there is no way to distinguish between a blank and an 'e'.

And by 'e' here I mean just 'e', not something like `1e6`. 'e' by itself isn't a valid value but we can't tell that apart from a blank.

**Comment by lcd2yyz** [comment]
I think for the majority of the app developer and users, the likelihood of typing 'e' with the intention of using the mathematical constant is quite low. One can easily specify out the actual value of 2.71828 for example. I feel okay to look past it for now, and only address the problem if/when user reports issue.

**Comment by asbates** [comment]
The 'e' here is for scientific notation, not the constant 'e'. But yea if you're OK assuming NA is blank we can go with that.

**Comment by asbates** [comment]
Yes it is confusing. I guess that's the HTML/browser spec but I don't think it makes sense.
